src/windowing/window.hpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <GLFW/glfw3.h>
+#include <types.hpp>
+#include <ui/movable_rect.hpp>
+
+namespace nebula {
+  struct Window {
+  public:
+    /* Variables */
+    GLFWwindow* glfw_window; // GLFW window instance
+    i32 width; // window width
+    i32 height; // window height
+    void (*custom_render)(); // function that renders all custom objects
+
+    /**
+     * Constructor initializes glfw and creates new fullscreen window with
+     * callbacks.
+     * @param custom_render_func - function that renders all custom objects.
+     * ( Other then MR buttons and created by them rectangles. )
+     */
+    explicit Window(void (*custom_render_func)());
+
+    /**
+     * Destroys window and terminates glfw.
+     */
+    ~Window();
+
+    /* Methods */
+
+    /**
+     * Make this window current context and initializes glad.
+     * @return 0 on success else 1.
+     */
+    [[nodiscard]] u8 make_current_context() const;
+
+    /**
+     * Function to render all objects.
+     */
+    void render() const;
+
+  private:
+    /* Private Variables */
+
+    Vec2 last_mouse_position; // last mouse position
+    List<Movable_Rect> movable_rectangles; // All created rectangles/gates
+    Movable_Rect* currently_moved_rectangle; // Currently moved rect/gate
+
+    /* Private Methods */
+
+    /**
+     * Keyboard input handler.
```

Comment:
this comment does not introduce any new information, hence it's redundant and unnecessary